app/src/main/java/com/nononsenseapps/feeder/model/FeedParser.kt

```diff
@@ -204,6 +235,132 @@ class FeedParser(override val di: DI) : DIAware {
      */
     private suspend fun curl(url: URL) = client.curl(url)
 
+    private suspend fun parseNostrUri(adaptedNostrUri: String): Nip19Profile {
+        val nostrUri = adaptedNostrUri.removePrefix("https://njump.me/")
+        if (nostrUri.contains("@")) { // It means it is a Nip05 address
+            val rawString = nostrUri.removePrefix("nostr:")
+            val parsedNip5 = getNip05Profile(rawString)
+            val (pubkey, relays) = parsedNip5.publicKey() to parsedNip5.relays()
+            return Nip19Profile(pubkey, relays)
+        } else {
+            val parsedProfile = Nip21.parse(nostrUri).asEnum()
+            when (parsedProfile) {
+                is Nip21Enum.Pubkey -> return Nip19Profile(parsedProfile.publicKey)
+                is Nip21Enum.Profile -> return Nip19Profile(parsedProfile.profile.publicKey(), parsedProfile.profile.relays())
+                else -> throw NostrUriParserException(message = "Could not find the user's info: $nostrUri")
+            }
+        }
+    }
+
+    suspend fun getProfileMetadata(nostrUri: URL): Either<MetaDataParseError, AuthorNostrData> {
+        return Either.catching(
+            onCatch = { t -> MetaDataParseError(url = nostrUri.toString(), throwable = t) },
+        ) {
+            val possibleNostrProfile = parseNostrUri(nostrUri.toString())
+            val publicKey = possibleNostrProfile.publicKey()
+            val relayList =
+                possibleNostrProfile.relays()
+                    .takeIf {
+                        it.size < 4
+                    }.orEmpty()
```

Comment:
Why are you only using the list if it has less than 4 relays?

Reply:
Generally, connecting to a big number of relays takes more in terms of time and bandwidth. And since we are just fetching the author's information, it should not take long, and 3 relays are more than enough for that.